plugins/scaffolder-backend/src/scaffolder/stages/helpers.ts

```diff
@@ -37,22 +38,16 @@ export const parseLocationAnnotation = (
     );
   }
 
-  // split on the first colon for the protocol and the rest after the first split
-  // is the location.
-  const [protocol, location] = annotation.split(/:(.+)/) as [
-    ('file' | 'url')?,
-    string?,
-  ];
-
-  if (!protocol || !location) {
+  const { type, target } = parseLocationReference(annotation);
+  if (!type || !target) {
     throw new InputError(
-      `Failure to parse either protocol or location for entity: ${entity.metadata.name}`,
+      `Failure to parse either type or target for entity: ${entity.metadata.name}`,
     );
   }
 
   return {
-    protocol,
-    location,
+    protocol: type as 'file' | 'url',
```

Comment:
yeah this isn't pretty, but it is equal to what as there before and this is code that's being deleted soon